base/migrator/migrator.Job.yaml

```diff
@@ -0,0 +1,56 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: migrator
+  annotations:
+    description: Database migrator for Postgres databases
+  labels:
+    app.component.kubernetes/component: migrator
+    deploy: sourcegraph
+spec:
+  template:
+    spec:
+      containers:
+      - name: migrator
+        image: "sourcegraph/migrator:3.35.1"
+        args: ["up"]
+        env:
+          - name: PGHOST
+            value: "pgsql"
+          - name: PGPORT
+            value: "5432"
+          - name: PGUSER
+            value: "sg"
+          - name: PGPASSWORD
+            value: "sg"
+          - name: PGDATABASE
+            value: "sg"
+          - name: PGSSLMODE
+            value: "disable"
+          - name: CODEINTEL_PGHOST
+            value: "codeintel-db"
+          - name: CODEINTEL_PGPORT
+            value: "5432"
+          - name: CODEINTEL_PGUSER
+            value: "sg"
+          - name: CODEINTEL_PGPASSWORD
+            value: "sg"
+          - name: CODEINTEL_PGDATABASE
+            value: "sg"
+          - name: CODEINTEL_PGSSLMODE
+            value: "disable"
+          - name: CODEINSIGHTS_PGHOST
+            value: "codeinsights-db"
+          - name: CODEINSIGHTS_PGPORT
+            value: "5432"
+          - name: CODEINSIGHTS_PGUSER
+            value: "postgres"
+          - name: CODEINSIGHTS_PGPASSWORD
+            value: "password"
+          - name: CODEINSIGHTS_PGDATABASE
+            value: "postgres"
+          - name: CODEINSIGHTS_PGSSLMODE
+            value: "disable"
+      restartPolicy: OnFailure
+  backoffLimit: 0
```

Comment:
What's the reasoning behind this backoffLimit? If a failure happens, it will need manual intervention to resolve even if it's a temporary problem - might not be a problem, just curious.

Reply:
Actually, it could be a big problem when we're talking about setting up a new Sourcegraph deployment. The first time any of this runs it could be several minutes before the databases are available, and in the meantime I assume this job will fail because it can't connect to them. At that point the customer will have a broken Sourcegraph and will have to re-run the kubectl-apply-all to trigger the job again.

Since Kubernetes doesn't have the idea of dependencies I'm not sure what a good option would be. An initContainer that waits until those connections are healthy, perhaps?